### ISSUE TITLE
XDCC bug when trying to download 2GB+ items

### ISSUE THREAD

**Issue by killerwife** [hexchat/hexchat#1556]
When trying to download 2GB+ files on windows using XDCC from various servers, the download crashed on about 2.0-2.1 GB. When I turned on mIRC and resumed the same download it worked perfectly fine (the file wasnt even corrupted, just not completely downloaded). I used both 2.10.2 and 2.11.0 version for testing, same issue with both. When the download fails and you try to resume, it writes "dont have access to file" type of message, while the file is there. Checked for folder permissions and everything was in order.

**Comment by Arnavion** [comment]
2.10.2 will definitely behave strangely. I'm able to download 4+ GiB files just fine with 2.11.0. #382 was fixed last year and it should work fine in 2.11.0 test builds.

- Is there a particular 2+ GiB file / XDCC server that it happens with or all of them?

- If it does happen for a file, are you never unable to resume it from HC no matter how many times you try? What's the exact error message.

- When did you download the 2.11.0 build you used?

- 32- or 64-bit HC? Which version of Windows?

**Comment by killerwife** [comment]
No, tested it on several servers/files using hexchat. My windows/hexchat is 64-bit, windows 10. The 2.11.0 was from this link https://dl.hexchat.net/hexchat/testing/
Should there be a newer version, let me know, ill test it.
Exact message is Cannot access file "filepath" the file does not exist. after it reaches the critical 2GB mark. (yet the file exists, even VLC can play it, and mIRC can resume it)

**Comment by Arnavion** [comment]
I asked you *when* you downloaded the 2.11.0 build, not *where* you downloaded it from.

**Comment by Arnavion** [comment]
Also, what is the full path of the file when you download it?

**Comment by killerwife** [comment]
"C:\Users\Killerwife\Downloads\[NoobSubs] Psycho-Pass The Movie (720p Blu-ray 8bit AAC).mp4"
And it was today, and hour before this bug report. And its important where it was from, since i couldve accessed someones more recent 2.11.0 since as far as i know, the definitive version is not out yet. (and the timestamps are kinda slightly older on those files)

**Comment by Arnavion** [comment]
>If it does happen for a file, are you never unable to resume it from HC no matter how many times you try?

You didn't answer this question.

>"C:\Users\Killerwife\Downloads[NoobSubs] Psycho-Pass The Movie (720p Blu-ray 8bit AAC).mp4"

Are you sure it's that? It's missing a `\` after `Downloads`

**Comment by killerwife** [comment]
Sry, i managed to make a typo while editing. I am not able to resume it from HC no matter how/what I do. 
If it doesnt throw the error message mentioned, the IRC bot replies with "connection closed" message. This happens when I try to establish the connection to finish/resume downloading after restarting hexchat (essentially making a new XDCC request, but it works this way for smaller files even in HC), while the "cannot access file" message pops out when I try to restart/resume download if I dont restart hexchat.

**Comment by Arnavion** [comment]
I can't help you if you don't read my comments completely and answer all my questions...

**Comment by killerwife** [comment]
I meant that the missing \ was a typo. Its there in the complete path.

**Comment by Arnavion** [comment]
What's the output of `/SET dcc_dir` ? Does it have a space at the end?

**Comment by killerwife** [comment]
Negative, no space at the end.

**Comment by Arnavion** [comment]
Two questions have two answers.

**Comment by killerwife** [comment]
The string is the output of /set dcc_dir, what else do you want. As i said it has no space at the end.

**Comment by Arnavion** [comment]
Last try: What's the output of `/SET dcc_dir` ?

**Comment by killerwife** [comment]
dcc_dir......................: C:\Users\Killerwife\Downloads

**Comment by killerwife** [comment]
I can give you a screenshot if you dont believe me.

**Comment by beep4beep** [comment]
Sorry if I resurrected a zombie, but the mentioned problem fits mine as well.

It seems more XDCCl related (maybe a "keep alive signal" that is somehow missed?). Meaning my downloads over XDCC will fail after reaching aprox 2GB with 3 different clients (my favorite hexchat, kvirc and adiIRC). Downloads over browsers or for example torrents are not effected.  And I tried over a dozen different bots.

Really weird: I can resume the download for the very next 2GB and it will fail aprox at 4GB and so on. Did a test on a 18GB file and it's always seems to drop the connection every 2GB on XDCC. And I'm pretty sure that other users with a similar rig are not effected. The error message points to a user interaction, but I didn't hit any key or button. 

win10 64 bit, all updates, NTFS Filesystem. HC 2.14.2 64bit (tried the 32bit for kvirc, the 64 bit for adiIRC)
Error message on HC: only the "download failed"
from the bot:  ** Closing Connection: Unable to transfer data (Connection reset by peer) 
/set dcc_dir returns:
dcc_dir......................: F:\IRCtemp
Which is correct, since unfinished files are set to that directory. 

The problem feels similar to the chain of events when a bot tries to transfer a file which is not accepted by the client (because there is already a finished file with the same name in that directory). But if that was the problem, why does the download work for every 2GB per request/file? And I checked, there was no file in the finished directory or a finished file in the temp directory that fits that particular pattern.
